### PR TITLE
move to new github actions config

### PR DESCRIPTION
**Pull request by hez**

### CODE REVIEW

File: .github/workflows/push.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+on: push
+name: Continuous Integration
```
Comment (stewart) [inline .github/workflows/push.yml]: Can we break these sections apart, and put the `name` at the top?

Comment (hez) [reply .github/workflows/push.yml]: We might be able to, the creation of the image to run them in is the expensive part and as far as I can tell we will lose the caching of that extending the run time of these checks quite a bit.

Comment (hez) [reply .github/workflows/push.yml]: The new details view (1 click away) is a lot better in the new github actions, that might be good enough?

Comment (stewart) [reply .github/workflows/push.yml]: I don't follow - I'm just looking to see if we can improve the readability of the yaml slightly:

```yaml
name: "Continuous Integration"

on: push

jobs:
  [...]
```

Comment (hez) [reply .github/workflows/push.yml]: done and done 👍

**Pull request merged**